lib/galaxy/auth/providers/ldap_ad.py

```diff
@@ -205,12 +219,32 @@ def authenticate(self, email, username, password, options):
             if not self._authenticate(params, options):
                 return failure_mode, '', ''
 
+        # check whether the user is a member of a specified group/domain/...
+        if 'search-memberof-filter' in options:
+            search_filter = _get_subs(options, 'search-memberof-filter', params)
+            if search_filter not in params['memberOf']:
+                return failure_mode, '', ''
+
         attributes = {}
         if self.auto_create_roles_or_groups:
             attributes['roles'] = params[self.role_search_option]
+
+        if string_as_bool(options.get('ldap-autogenerate', False)):
+            if str(username) == 'None':
+                register_username = params['displayName']
+                register_email = _get_subs(options, 'auto-register-email', params)
+            else:
+                register_username = _get_subs(options, 'auto-register-username', params)
+                register_email = params['mail']
+        else:
+	    register_email = _get_subs(options, 'auto-register-email', params)
+            register_username = _get_subs(options, 'auto-register-username', params)
+
+        register_username = _clean_register_username(register_username)
```

Comment:
Is it really a good idea to modify the username? I would have assumed that the username is checked letter by letter by the auth provider...

Reply:
Done for constancy with the rule display into account creation panel : "Public names must be at least three characters in length and contain only lower-case letters, numbers, dots, underscores, and dashes ('.', '_', '-').".  This is the public name used by Galaxy ( in case off auto-registration) and return by the provider, not the username used by the provider to check the credentials (this one remained unmodified) .